verifiers/interfaceValidator.js

```diff
@@ -44,17 +44,52 @@ function getRamlFilePath(filePath) {
 }
 
 // Get all of the methods defined in a raml schema file.
-// TODO: update how the names are assembled to handle multiple parts and to ignore
-// parameter values.
 function getSchemaMethods(ramlPath) {
+
+  function addMethod(methodName) {
+    if(ramlMethods.indexOf(methodName) === -1) {
+      ramlMethods.push(methodName);
+    }
+  }
+
   var ramlMethods = [];
 
   if(ramlPath) {
-    raml.loadApiSync(ramlPath).resources().forEach(resource => {
-      resource.methods().forEach(method => {
-        ramlMethods.push(method.method() + resource.completeRelativeUri().substring(1));
-      });
+    var resources = [];
+    raml.loadApiSync(ramlPath).resources().forEach(r => {
+      resources.push(r);
     });
+
+    while(resources.length > 0) {
```

Comment:
I'd recommend adding comments for each section with examples of what is being looked at and expected